ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/MRCompactor.java

```diff
@@ -261,7 +285,7 @@ public void run(HiveConf conf, String jobName, Table t, Partition p, StorageDesc
 
     StringableList dirsToSearch = new StringableList();
     Path baseDir = null;
-    if (ci.isMajorCompaction()) {
+    if (CompactionType.MAJOR.equals(ci.type)) {
```

Comment:
why is this change?

Reply:
There were several parts of the code which worked in a way liek below:
`
 if (!ci.isMajorCompaction()) {
    //Assuming it is minor compaction but it is no longer true!!!
}
`
To enforce re-checking all occurences and to prevent adding such wrong conditions in the future I decided to remove the function.

Reply:
````
boolean isMajorCompaction(ci){
return CompactionType.MAJOR.equals(ci.type)
}
````
i think that is a useful helper function, but up to you

Reply:
O, I re-added and also added isRebalanceCompaction()